### ISSUE TITLE
Creating new jablkomod instances dynamically

### ISSUE THREAD

**Issue by ccoverstreet** [ccoverstreet/Jablko#43]
Right now, all jablkomods to be used in during runtime are configured and instantiated at startup. What is needed is a way to create new jablkomod instances without having to restart. Here are some ideas for the general program flow. These functions should be implemented as JablkoModuleHolder method. 

1. Client in an admin page sends a request to install a jablkomod using a source directory and title (maybe more arguments)
    ```go
    func InstallJablkoMod(sourceDir string, title string)
    ```
2. Check if plugin file is already present
3. If not, go through download process already utilized in `Initialize()`.
4. Build newly downloaded module.
5. Run initialize with but provide `nil` as the configData argument. 
    - Modules should handle a `nil` case by creating an instance with a default configuration
6. This instance should then be added to the ModMap struct under a new ID generated by a nanosecond timestamp.
    - This ID should be generated as an string with no numbers or special characters as this gets substituted into the HTML template
7. The generated ID should then be appended to the end of the JablkoModuleHolder.Order.
8. Once initialization is complete, MainApp.SyncConfig() should be called to write the new config file

Implementing this feature should finish a majority of the jablkomods implementation and may warrant creating a 0.3.0 release. Fully updated documentation regarding modules should be paired with the next release.

**Comment by ccoverstreet** [comment]
Documentation update moved to separate issue #44.

**Comment by ccoverstreet** [comment]
closes #37

**Comment by ccoverstreet** [comment]
#46 is also related

**Comment by ccoverstreet** [comment]
Documentation of downloading system and jablkomod go.mod file setup are needed

**Comment by ccoverstreet** [comment]
Next jablkomods-install pull request should finish implementing this.
